### ISSUE TITLE
 Don't know how to find reference variables in Ecto query in from list in query

### ISSUE THREAD

**Issue by tonidy** [KronicDeth/intellij-elixir#2088]
# Version

11.13.0

# What I was doing

I opened elixir file with code like this:

### Excerpt

```
[shipment, job]
```

From: `/path/to/file.ex:42`
To: `/path/to/file/file.ex:42`

### Element Class Name

```
org.elixir_lang.psi.impl.ElixirAccessExpressionImpl
```
Here is code:
```
defp query(query, date, orentiation) when not is_nil(date) do # line:40
    case orentiation do # line:41
      :lte -> # line:42 -> error happened here
        from [shipment, job] in query,
          where: fragment("(? something')::date", job.date) <= ^date

```

# Exceptions

### Stacktrace

```
java.lang.Throwable: Don't know how to find reference variables in Ecto query
	at org.elixir_lang.errorreport.Logger.error(Logger.java:42)
	at org.elixir_lang.ecto.Query.executeOnInReference(Query.kt:151)
	at org.elixir_lang.ecto.Query.executeOnIn(Query.kt:139)
	at org.elixir_lang.ecto.Query.executeOnFrom(Query.kt:125)
	at org.elixir_lang.ecto.Query.treeWalkUp(Query.kt:45)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.kt:245)
	at org.elixir_lang.psi.scope.Variable.execute(Variable.kt:64)
	at org.elixir_lang.psi.impl.ProcessDeclarationsImpl.processDeclarations(ProcessDeclarationsImpl.kt:145)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.processDeclarations(ElixirPsiImplUtil.java:581)
	at org.elixir_lang.psi.impl.ElixirUnmatchedUnqualifiedNoParenthesesCallImpl.processDeclarations(ElixirUnmatchedUnqualifiedNoParenthesesCallImpl.java:157)
	at com.intellij.psi.util.PsiTreeUtil.treeWalkUp(PsiTreeUtil.java:1218)
	at org.elixir_lang.psi.scope.variable.MultiResolve$Companion.resolveResultList(MultiResolve.kt:123)
	at org.elixir_lang.psi.scope.variable.MultiResolve$Companion.resolveResultList(MultiResolve.kt:108)
	at org.elixir_lang.reference.resolver.Callable.resolveUnqualified(Callable.kt:68)
	at org.elixir_lang.reference.resolver.Callable.resolveInScope(Callable.kt:53)
	at org.elixir_lang.reference.resolver.Callable.resolve(Callable.kt:44)
	at org.elixir_lang.reference.resolver.Callable.resolveAll(Callable.kt:40)
	at org.elixir_lang.reference.resolver.Callable.resolvePreferred(Callable.kt:30)
	at org.elixir_lang.reference.resolver.Callable.resolve(Callable.kt:26)
	at org.elixir_lang.reference.resolver.Callable.resolve(Callable.kt:22)
	at com.intellij.psi.impl.source.resolve.ResolveCache.lambda$resolve$1(ResolveCache.java:154)
	at com.intellij.openapi.util.Computable.get(Computable.java:18)
	at com.intellij.psi.impl.source.resolve.ResolveCache.lambda$loggingResolver$4(ResolveCache.java:260)
	at com.intellij.openapi.util.Computable.get(Computable.java:18)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolve(ResolveCache.java:238)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolve(ResolveCache.java:154)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:169)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:162)
	at org.elixir_lang.reference.ResolveWithCachingComputable.compute(Callable.kt:595)
	at org.elixir_lang.reference.ResolveWithCachingComputable.compute(Callable.kt:587)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:841)
	at org.elixir_lang.reference.CallableKt.resolveWithCaching(Callable.kt:585)
	at org.elixir_lang.reference.CallableKt.access$resolveWithCaching(Callable.kt:1)
	at org.elixir_lang.reference.Callable.multiResolve(Callable.kt:107)
	at org.elixir_lang.ReferenceKt.safeMultiResolve(reference.kt:15)
	at org.elixir_lang.annotator.Callable$annotate$1.visitPlainCall(Callable.kt:97)
	at org.elixir_lang.annotator.Callable$annotate$1.visitNonModuleAttributeCall(Callable.kt:87)
	at org.elixir_lang.annotator.Callable$annotate$1.visitCall(Callable.kt:47)
	at org.elixir_lang.annotator.Callable$annotate$1.visitElement(Callable.kt:78)
	at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:273)
	at org.elixir_lang.psi.impl.ElixirUnmatchedUnqualifiedNoArgumentsCallImpl.accept(ElixirUnmatchedUnqualifiedNoArgumentsCallImpl.java:40)
	at org.elixir_lang.annotator.Callable.annotate(Callable.kt:39)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.runAnnotators(DefaultHighlightVisitor.java:134)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.visit(DefaultHighlightVisitor.java:114)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.runVisitors(GeneralHighlightingPass.java:335)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$collectHighlights$5(GeneralHighlightingPass.java:274)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:294)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$analyzeByVisitors$6(GeneralHighlightingPass.java:297)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.analyze(DefaultHighlightVisitor.java:94)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:297)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectHighlights(GeneralHighlightingPass.java:265)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectInformationWithProgress(GeneralHighlightingPass.java:211)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:84)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:56)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:414)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1078)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:407)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:705)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:647)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:63)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:406)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:382)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:174)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:183)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:380)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:188)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by KronicDeth** [comment]
I didn't know you could use lists as the the left operand to `in` for an existing query in `from`. TIL.

**Comment by KronicDeth** [comment]
This is fixed on `main`.  It was resolved by #2179.  It will be released in `12.0.0`.
